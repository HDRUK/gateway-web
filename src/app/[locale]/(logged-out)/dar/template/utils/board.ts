import { BoardSections, Board, Task } from "../components";
import { BOARD_SECTIONS } from "../constants";
import { getTasksByStatus } from "./tasks";

export const initializeBoard = (tasks: Task[]) => {
    const boardSections: BoardSections = {};

    Object.keys(BOARD_SECTIONS).forEach(boardSectionKey => {
        boardSections[boardSectionKey] = getTasksByStatus(
            tasks,
            boardSectionKey as Board
        );
    });

    return boardSections;
};

export const findBoardSectionContainer = (
    boardSections: BoardSections,
    id: string
) => {
    if (id in boardSections) {
        return id;
    }

    const container = Object.keys(boardSections).find(key =>
        boardSections[key].find(item => item.id === id)
    );
    return container;
};
