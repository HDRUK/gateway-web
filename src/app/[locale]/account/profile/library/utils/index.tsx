import { IconButton } from "@mui/material";
import { createColumnHelper } from "@tanstack/react-table";
import { LibraryListItem, SelectedLibrary } from "@/interfaces/Library";
import Link from "@/components/Link";
import StyledCheckbox from "@/components/StyledCheckbox";
import TooltipIcon from "@/components/TooltipIcon";
import { CheckIcon, DeleteForeverIcon } from "@/consts/icons";
import { RouteName } from "@/consts/routeName";

const columnHelper = createColumnHelper<LibraryListItem>();

const getColumns = ({
    handleSelect,
    handleRemove,
    selected,
    translations,
}: {
    handleSelect: (data: SelectedLibrary) => void;
    handleRemove: (id: string | number) => void;
    selected: SelectedLibrary;
    translations: { [id: string]: string };
}) => [
    columnHelper.display({
        id: "actions",
        meta: { isPinned: true },
        cell: ({ row }) => {
            const {
                id,
                datasetId,
                name,
                dataCustodianId,
                dataCustodian,
                darEnabled,
                darTemplatePublished,
                cohortEnabled,
            } = row.original;
            return (
                <div style={{ textAlign: "center" }}>
                    <StyledCheckbox
                        checked={selected[id]?.selected}
                        onChange={(_e, value) =>
                            handleSelect({
                                [id]: {
                                    selected: value,
                                    datasetId,
                                    name,
                                    teamId: dataCustodianId,
                                    teamName: dataCustodian,
                                    darEnabled,
                                    darTemplatePublished,
                                    cohortEnabled,
                                },
                            })
                        }
                        size="large"
                        sx={{ p: 0 }}
                        iconSx={{ mr: 0 }}
                    />
                </div>
            );
        },
        header: () => null,
        size: 43,
    }),

    columnHelper.display({
        id: "name",
        cell: ({ row: { original } }) => {
            const { datasetId, name } = original;
            const linkHref = `/${RouteName.DATASET_ITEM}/${datasetId}`;

            return (
                <Link href={linkHref}>
                    <div style={{ textAlign: "center" }}>{name}</div>
                </Link>
            );
        },
        header: () => <span>{translations.name}</span>,
        size: 150,
    }),

    columnHelper.display({
        id: "darEnabled",
        cell: ({
            row: {
                original: { darEnabled },
            },
        }) => (
            <div style={{ textAlign: "center" }}>
                {darEnabled ? <CheckIcon color="primary" /> : "-"}
            </div>
        ),
        header: () => (
            <TooltipIcon
                buttonSx={{ p: 0 }}
                size="small"
                label={translations.darEnabled}
                content={translations.darEnabled}
            />
        ),
        size: 100,
    }),

    columnHelper.display({
        id: "cohortEnabled",
        cell: ({
            row: {
                original: { cohortEnabled },
            },
        }) => (
            <div style={{ textAlign: "center" }}>
                {cohortEnabled ? <CheckIcon color="primary" /> : "-"}
            </div>
        ),
        header: () => <span>{translations.cohortEnabled}</span>,
        size: 100,
    }),

    columnHelper.display({
        id: "dataCustodian",
        cell: ({
            row: {
                original: { dataCustodian },
            },
        }) => <div style={{ textAlign: "center" }}>{dataCustodian}</div>,
        header: () => <span>{translations.dataCustodian}</span>,
        size: 100,
    }),

    columnHelper.display({
        id: "entityType",
        cell: ({
            row: {
                original: { entityType },
            },
        }) => <div style={{ textAlign: "center" }}>{entityType}</div>,
        header: () => <span>{translations.entityType}</span>,
        size: 100,
    }),

    columnHelper.display({
        id: "delete",
        cell: ({ row: { original } }) => {
            return (
                <div style={{ textAlign: "center" }}>
                    <IconButton onClick={() => handleRemove(original.id)}>
                        <DeleteForeverIcon color="primary" />
                    </IconButton>
                </div>
            );
        },
        header: () => null,
        size: 43,
    }),
];

export { getColumns };
