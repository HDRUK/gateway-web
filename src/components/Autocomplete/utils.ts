import { ValueType } from "./Autocomplete";

const getChipLabel = (
    options: { value: string | number; label: string }[],
    value: ValueType
) => {
    if (typeof value === "string") {
        return value;
    }

    return options.find(option => option.value === value)?.label;
};

export { getChipLabel };
