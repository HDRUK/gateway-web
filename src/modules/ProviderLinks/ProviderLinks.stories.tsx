import type { Meta, StoryObj } from "@storybook/react";
import ProviderLinks from "./ProviderLinks";

const meta: Meta<typeof ProviderLinks> = {
    component: ProviderLinks,
    tags: ["autodocs"],
};

export default meta;

type Story = StoryObj<typeof ProviderLinks>;

export const Default: Story = {};
