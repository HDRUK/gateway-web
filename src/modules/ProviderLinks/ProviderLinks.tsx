import { useTranslations } from "next-intl";
import Image from "next/image";
import { usePathname } from "next/navigation";
import Box from "@/components/Box";
import Link from "@/components/Link";
import apis from "@/config/apis";
import { colors } from "@/config/theme";
import { RouteName } from "@/consts/routeName";

interface LinkItem {
    label: string;
    href: string;
    image: string;
}

const ProviderLinks = () => {
    const t = useTranslations("modules");
    const pathname = usePathname();

    let redirectPath = "";

    if (
        pathname?.includes(`/${RouteName.ABOUT}/${RouteName.COHORT_DISCOVERY}`)
    ) {
        redirectPath = `?redirect=/${RouteName.ABOUT}/${RouteName.COHORT_DISCOVERY_REQUEST}`;
    }

    const providerLinks: LinkItem[] = [
        {
            label: t("dialogs.ProvidersDialog.socialProviders.google"),
            href: `${apis.authGoogleV1Url}${redirectPath}`,
            image: "google-logo.png",
        },
        {
            label: t("dialogs.ProvidersDialog.socialProviders.linkedIn"),
            href: `${apis.authLinkedinV1Url}${redirectPath}`,
            image: "linkedIn-logo.png",
        },
        {
            label: t("dialogs.ProvidersDialog.socialProviders.azure"),
            href: `${apis.authAzureV1Url}${redirectPath}`,
            image: "microsoft-logo.png",
        },
    ];
    return (
        <Box
            sx={{
                padding: 0,
                display: "grid",
                columnGap: 3,
                rowGap: 2,
                gridTemplateColumns: {
                    mobile: "repeat(1, 1fr)",
                    tablet: "repeat(2, 1fr)",
                },
                marginTop: 3,
                marginBottom: 3,
            }}>
            {providerLinks.map(link => (
                <Link
                    href={link.href}
                    key={link.image}
                    prefetch={false}
                    underline="none"
                    sx={{ color: colors.grey700 }}>
                    <Box
                        sx={{
                            border: `solid 1px ${colors.grey400}`,
                            borderRadius: "4px",
                            padding: "7px",
                            display: "flex",
                            gap: 2,
                        }}>
                        <Image
                            src={`/images/logos/${link.image}`}
                            alt={link.label}
                            width="20"
                            height="20"
                        />
                        Sign in with {link.label}
                    </Box>
                </Link>
            ))}
        </Box>
    );
};

export default ProviderLinks;
