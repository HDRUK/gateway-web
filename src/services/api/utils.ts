import { NotificationOptions } from "@/interfaces/Api";
import { AxiosResponse } from "axios";
import { Error } from "@/interfaces/Error";
import notificationService from "../notification";

interface ErrorNotificationProps {
    props: NotificationOptions;
    errorResponse: AxiosResponse<Error>;
    method: "delete" | "post" | "put" | "get";
}

const errorNotification = ({
    errorResponse,
    method,
    props,
}: ErrorNotificationProps) => {
    const { t, i18n, ...notificationProps } = props;
    const { data, status } = errorResponse || {};

    const fallbackTitle = i18n.exists(`api:common.error.status.${status}`)
        ? t(`api:common.error.status.${status}`)
        : "There has been an error";

    const title = data?.title || fallbackTitle;
    const message =
        data?.message ||
        t(`api:common.error.${method}.message`, {
            item: props.itemName || "Item",
        });

    notificationService.apiError(message, {
        title,
        message,
        errors: data?.errors,
        ...notificationProps,
    });
};

interface SuccessNotificationProps {
    props: NotificationOptions;
    method: "delete" | "post" | "put";
}

const successNotification = ({ props, method }: SuccessNotificationProps) => {
    const { t, i18n, ...notificationProps } = props;
    const customMessage = `api:${props.localeKey}.success.${method}.message`;
    const shouldOverideMessage = i18n.exists(customMessage);

    const message = shouldOverideMessage
        ? t(customMessage)
        : t(`api:common.success.${method}.message`, {
              item: props.itemName || "Item",
          });

    notificationService.success(message, { ...notificationProps });
};

export { errorNotification, successNotification };
