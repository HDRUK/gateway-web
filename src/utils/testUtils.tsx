import React, { ReactNode } from "react";
import {
    render,
    renderHook,
    RenderHookOptions,
    RenderOptions,
    RenderResult,
    RenderHookResult,
} from "@testing-library/react";
import { SWRConfig } from "swr";
import { CacheProvider, ThemeProvider } from "@emotion/react";
import ActionBarProvider from "@/providers/ActionBarProvider";
import DialogProvider from "@/providers/DialogProvider";
import theme from "@/config/theme";
import createEmotionCache from "@/config/createEmotionCache";

const clientSideEmotionCache = createEmotionCache();

const Wrapper = ({ children }: { children: ReactNode }) => {
    return (
        <SWRConfig
            value={{
                provider: () => new Map(),
            }}>
            <CacheProvider value={clientSideEmotionCache}>
                <ThemeProvider theme={theme}>
                    <ActionBarProvider>
                        <DialogProvider>{children}</DialogProvider>
                    </ActionBarProvider>
                </ThemeProvider>
            </CacheProvider>
        </SWRConfig>
    );
};

interface OptionProps
    extends RenderOptions<
        typeof import("@testing-library/dom/types/queries"),
        HTMLElement,
        HTMLElement
    > {
    wrapperProps: Record<string, unknown>;
}

const customRender = (
    ui: React.ReactElement<
        unknown,
        string | React.JSXElementConstructor<unknown>
    >,
    options?: OptionProps
): RenderResult => {
    const { wrapperProps, ...rest } = options || {};

    return render(ui, {
        wrapper: props => <Wrapper {...props} {...wrapperProps} />,
        ...rest,
    });
};

interface OptionHookProps
    extends RenderHookOptions<
        unknown,
        typeof import("@testing-library/dom/types/queries"),
        HTMLElement,
        HTMLElement
    > {
    wrapperProps: Record<string, unknown>;
}

const customRenderHook = (
    ui: (initialProps: unknown) => unknown,
    options?: OptionHookProps
): RenderHookResult<unknown, unknown> => {
    const { wrapperProps, ...rest } = options || {};
    return renderHook(ui, {
        wrapper: props => <Wrapper {...props} {...wrapperProps} />,
        ...rest,
    });
};

export * from "@testing-library/react";

export { customRenderHook as renderHook };
export { customRender as render };
