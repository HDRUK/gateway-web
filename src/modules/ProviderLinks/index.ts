import ProviderLinks from "./ProviderLinks";

export default ProviderLinks;
