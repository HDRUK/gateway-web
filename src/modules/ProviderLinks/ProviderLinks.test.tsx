import React from "react";
import ProviderLinks from "@/modules/ProviderLinks";
import { render, screen } from "@/utils/testUtils";

describe("ProviderLinks", () => {
    it("should match snapshot", async () => {
        const wrapper = render(<ProviderLinks />);

        expect(wrapper.container).toMatchSnapshot();
    });

    it("should render component", async () => {
        render(<ProviderLinks />);

        expect(
            screen.getByAltText("dialogs.ProvidersDialog.socialProviders.azure")
        ).toBeInTheDocument();
        expect(
            screen.getByAltText(
                "dialogs.ProvidersDialog.socialProviders.linkedIn"
            )
        ).toBeInTheDocument();
        expect(
            screen.getByAltText(
                "dialogs.ProvidersDialog.socialProviders.google"
            )
        ).toBeInTheDocument();
    });
});
