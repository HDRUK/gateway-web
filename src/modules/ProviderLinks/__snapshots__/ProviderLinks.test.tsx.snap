// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProviderLinks should match snapshot 1`] = `
<div>
  <div
    class="MuiBox-root mui-style-6bfkjt"
  >
    <div
      class="MuiBox-root mui-style-1t6hs1a"
    >
      <img
        alt="dialogs.ProvidersDialog.socialProviders.azure"
        data-nimg="1"
        decoding="async"
        height="20"
        loading="lazy"
        src="/_next/image?url=%2Fimages%2Flogos%2Fmicrosoft-logo.png&w=48&q=75"
        srcset="/_next/image?url=%2Fimages%2Flogos%2Fmicrosoft-logo.png&w=32&q=75 1x, /_next/image?url=%2Fimages%2Flogos%2Fmicrosoft-logo.png&w=48&q=75 2x"
        style="color: transparent;"
        width="20"
      />
      <a
        class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways mui-style-1qmnj94-MuiTypography-root-MuiLink-root"
        href="http://localhost:8000/api/v1/auth/azure"
      >
        dialogs.ProvidersDialog.socialProviders.azure
      </a>
    </div>
    <div
      class="MuiBox-root mui-style-1t6hs1a"
    >
      <img
        alt="dialogs.ProvidersDialog.socialProviders.linkedIn"
        data-nimg="1"
        decoding="async"
        height="20"
        loading="lazy"
        src="/_next/image?url=%2Fimages%2Flogos%2FlinkedIn-logo.png&w=48&q=75"
        srcset="/_next/image?url=%2Fimages%2Flogos%2FlinkedIn-logo.png&w=32&q=75 1x, /_next/image?url=%2Fimages%2Flogos%2FlinkedIn-logo.png&w=48&q=75 2x"
        style="color: transparent;"
        width="20"
      />
      <a
        class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways mui-style-1qmnj94-MuiTypography-root-MuiLink-root"
        href="http://localhost:8000/api/v1/auth/linkedin"
      >
        dialogs.ProvidersDialog.socialProviders.linkedIn
      </a>
    </div>
    <div
      class="MuiBox-root mui-style-1t6hs1a"
    >
      <img
        alt="dialogs.ProvidersDialog.socialProviders.google"
        data-nimg="1"
        decoding="async"
        height="20"
        loading="lazy"
        src="/_next/image?url=%2Fimages%2Flogos%2Fgoogle-logo.png&w=48&q=75"
        srcset="/_next/image?url=%2Fimages%2Flogos%2Fgoogle-logo.png&w=32&q=75 1x, /_next/image?url=%2Fimages%2Flogos%2Fgoogle-logo.png&w=48&q=75 2x"
        style="color: transparent;"
        width="20"
      />
      <a
        class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways mui-style-1qmnj94-MuiTypography-root-MuiLink-root"
        href="http://localhost:8000/api/v1/auth/google"
      >
        dialogs.ProvidersDialog.socialProviders.google
      </a>
    </div>
  </div>
</div>
`;
